pytket/phir/phirgen_parallel.py

```diff
@@ -31,25 +31,26 @@
 
 
 def exec_order_preserved_helper(
-    ordered_dict: OrderedDict["UnitID", int], group_number: int, qubit_last_group: int
+    group_exec_order: list[int], group_number: int, qubit_last_group: int
 ) -> bool:
     """A helper to determine whether order is preserved when adding qubits to groups."""
     # determine whether the current group number is later in execution
     # than the last group in which a qubit was used
+    if (group_number not in group_exec_order) or (
+        qubit_last_group not in group_exec_order
+    ):
+        return True
     group_eligible = group_number > qubit_last_group
     if not group_eligible:
         return False
-    for group in ordered_dict.values():
-        if group == qubit_last_group:
-            order_preserved = False
-            break
-        if group == group_number:
-            order_preserved = True
-            break
-    return order_preserved
+    # if the group that the qubit is eligible for is later in the exec_order list than
+    # the last group in which it was used, it can be parallelized
+    return group_exec_order.index(group_number) > group_exec_order.index(
+        qubit_last_group
+    )
 
 
-def process_sub_commands(
+def process_sub_commands(  # noqa: PLR0914
```

Comment:
```suggestion
def process_sub_commands(
```
I think can eliminate the variable `order_preserved` below.